analyzers/conversion_overflow.go

```diff
@@ -140,9 +140,22 @@ func parseIntType(intType string) (integer, error) {
 	var max uint
 
 	if signed {
-		max = (1 << uint(intSize-1)) - 1
+		shiftAmount := intSize - 1
+
+		// Perform a bounds check
+		if shiftAmount < 0 {
+			return integer{}, fmt.Errorf("invalid shift amount: %d", shiftAmount)
+		}
```

Comment:
I've added basic support for EQL and NEQ. 

There are still annoyances like this one where I cannot subtract from a value that I have already validated without a redundant check, even though the conversion is completely safe in this case.